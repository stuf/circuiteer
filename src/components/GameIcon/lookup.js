export const synonyms = {
  sun: 'SolarPanel',
  wind: 'WindTurbine',
  always: 'Generator',
  smallGenerator: 'Generator',
  smallWindTurbine: 'WindTurbine',
  smallSolarPanel: 'SolarPanel',
  mediumGenerator: 'Generator',
  mediumShredder: 'Shredder',
  mediumWindTurbine: 'WindTurbine',
  mediumSolarPanel: 'SolarPanel',
  fieldShelter: 'Shelter',
  shelter: 'Shelter',
};

export const tier = {
  1: 'Small',
  2: 'Medium',
  3: 'Large',
  4: 'ExtraLarge',
};
