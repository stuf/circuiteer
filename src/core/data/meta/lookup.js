import { merge } from 'ramda';
import * as mods from 'core/data/modules';

export const lookup = {
  module: Object.values(mods)
    .flat()
    .reduce((o, mod) => merge(o, { [mod.id]: mod }), {}),
};
