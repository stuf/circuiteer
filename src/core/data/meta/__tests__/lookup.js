import { lookup } from '../lookup';

describe('lookup', () => {
  test('module', () => {
    expect(lookup.module).toMatchSnapshot();
  });
});
