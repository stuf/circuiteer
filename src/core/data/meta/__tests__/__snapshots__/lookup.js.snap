// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lookup module 1`] = `
Object {
  "atmosphericCondenser": Object {
    "id": "atmosphericCondenser",
    "power": -20,
    "recipe": Object {
      "glass": 1,
      "iron": 1,
      "plastic": 1,
    },
    "shortId": "ac",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 2200,
  },
  "autoArm": Object {
    "id": "autoArm",
    "power": -1,
    "recipe": Object {
      "aluminum": 1,
      "graphite": 1,
    },
    "shortId": "aa",
    "size": Array [
      2,
      2,
    ],
    "tier": 2,
  },
  "autoExtractor": Object {
    "id": "autoExtractor",
    "power": -8,
    "recipe": Object {
      "rubber": 1,
      "steel": 2,
      "tungstenCarbide": 1,
    },
    "shortId": "ae",
    "size": Array [
      8,
      8,
    ],
    "tier": 4,
    "unlockCost": 7500,
  },
  "buggy": Object {
    "id": "buggy",
    "power": -0.031,
    "recipe": Object {
      "aluminum": 1,
      "compound": 1,
    },
    "shortId": "b",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 1500,
  },
  "chemistryLab": Object {
    "id": "chemistryLab",
    "power": -10,
    "recipe": Object {
      "ceramic": 1,
      "glass": 1,
      "tungsten": 1,
    },
    "shortId": "cl",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 1600,
  },
  "exoRequestPlatform": Object {
    "id": "exoRequestPlatform",
    "power": null,
    "recipe": Object {
      "ceramic": 1,
      "resin": 2,
    },
    "shortId": "erp",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 1000,
  },
  "extraLargeShredder": Object {
    "id": "extraLargeShredder",
    "power": -10,
    "recipe": Object {
      "steel": 2,
      "tungstenCarbide": 2,
    },
    "shortId": "xls",
    "size": Array [
      8,
      8,
    ],
    "tier": 4,
  },
  "extraLargeStorage": Object {
    "id": "extraLargeStorage",
    "recipe": Object {
      "ceramic": 2,
      "iron": 2,
    },
    "shortId": "xlst",
    "size": Array [
      8,
      8,
    ],
    "tier": 4,
    "unlockCost": 2000,
  },
  "fieldShelter": Object {
    "id": "fieldShelter",
    "power": 0.5,
    "powerType": "always",
    "recipe": Object {
      "graphene": 1,
      "silicone": 1,
    },
    "shortId": "fs",
    "size": Array [
      2,
      2,
    ],
    "tier": 2,
  },
  "landingPad": Object {
    "id": "landingPad",
    "recipe": Object {
      "aluminum": 1,
      "ceramic": 2,
    },
    "shortId": "lp",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 750,
  },
  "largeActiveStorage": Object {
    "id": "largeActiveStorage",
    "power": null,
    "recipe": Object {
      "aluminum": 1,
      "resin": 1,
      "zinc": 1,
    },
    "shortId": " las",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 2000,
  },
  "largePrinter": Object {
    "id": "largePrinter",
    "power": -5,
    "recipe": Object {
      "compound": 3,
    },
    "shortId": "lp",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": null,
  },
  "largeShredder": Object {
    "id": "largeShredder",
    "power": -7.5,
    "recipe": Object {
      "iron": 2,
      "tungstenCarbide": 1,
    },
    "shortId": "ls",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 2500,
  },
  "largeSolarPanel": Object {
    "id": "largeSolarPanel",
    "power": 8,
    "powerType": "sun",
    "recipe": Object {
      "aluminumAlloy": 1,
      "copper": 1,
      "glass": 1,
    },
    "shortId": "lsp",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
  },
  "largeStorage": Object {
    "id": "largeStorage",
    "power": null,
    "recipe": Object {
      "ceramic": 3,
    },
    "shortId": "ls",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 2000,
  },
  "largeStorageSiloA": Object {
    "id": "largeStorageSiloA",
    "power": null,
    "recipe": Object {
      "aluminum": 2,
      "steel": 1,
    },
    "shortId": "lssa",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 5000,
  },
  "largeStorageSiloB": Object {
    "id": "largeStorageSiloB",
    "power": null,
    "recipe": Object {
      "steel": 3,
    },
    "shortId": "lssb",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 7500,
  },
  "largeWindTurbine": Object {
    "id": "largeWindTurbine",
    "power": 10,
    "powerType": "wind",
    "recipe": Object {
      "aluminumAlloy": 1,
      "ceramic": 1,
      "glass": 1,
    },
    "shortId": "lwt",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
  },
  "mediumBattery": Object {
    "capacity": 512,
    "id": "mediumBattery",
    "power": 5,
    "powerType": "powered",
    "recipe": Object {
      "lithium": 1,
      "zinc": 1,
    },
    "shortId": "mb",
    "size": Array [
      2,
      2,
    ],
    "tier": 2,
  },
  "mediumGenerator": Object {
    "id": "mediumGenerator",
    "power": 9,
    "powerType": "powered",
    "recipe": Object {
      "aluminum": 1,
      "tungsten": 1,
    },
    "shortId": "mg",
    "size": Array [
      2,
      2,
    ],
    "tier": 2,
  },
  "mediumPrinter": Object {
    "id": "mediumPrinter",
    "power": -2,
    "recipe": Object {
      "compound": 2,
    },
    "shortId": "mp",
    "size": Array [
      2,
      2,
    ],
    "tier": 2,
  },
  "mediumRover": Object {
    "id": "mediumRover",
    "power": -0.5,
    "recipe": Object {
      "plastic": 2,
      "rubber": 1,
    },
    "shortId": "mr",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 3750,
  },
  "mediumShredder": Object {
    "id": "mediumShredder",
    "power": -5,
    "recipe": Object {
      "iron": 2,
    },
    "shortId": "ms",
    "size": Array [
      2,
      2,
    ],
    "tier": 2,
  },
  "mediumSolarPanel": Object {
    "id": "mediumSolarPanel",
    "power": 4,
    "powerType": "sun",
    "recipe": Object {
      "copper": 1,
      "glass": 1,
    },
    "shortId": "msp",
    "size": Array [
      2,
      2,
    ],
    "tier": 2,
  },
  "mediumWindTurbine": Object {
    "id": "mediumWindTurbine",
    "power": 5,
    "powerType": "wind",
    "recipe": Object {
      "aluminum": 1,
      "ceramic": 1,
    },
    "shortId": "mwt",
    "size": Array [
      2,
      2,
    ],
    "tier": 2,
  },
  "portableOxygenator": Object {
    "id": "portableOxygenator",
    "power": -1,
    "recipe": Object {
      "nanoCarbonAlloy": 1,
    },
    "shortId": "po",
    "size": Array [
      1,
      1,
    ],
    "tier": 1,
  },
  "qtRtg": Object {
    "id": "qtRtg",
    "power": 1,
    "powerType": "always",
    "recipe": null,
    "shortId": "qr",
    "size": Array [
      1,
      1,
    ],
    "tier": 1,
  },
  "researchChamber": Object {
    "id": "researchChamber",
    "power": -2,
    "recipe": Object {
      "compound": 2,
      "resin": 1,
    },
    "shortId": "rc",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": null,
  },
  "rtg": Object {
    "id": "rtg",
    "power": 4,
    "powerType": "always",
    "recipe": Object {
      "lithium": 1,
      "nanoCarbonAlloy": 1,
    },
    "shortId": "rtg",
    "size": Array [
      2,
      2,
    ],
    "tier": 2,
  },
  "shelter": Object {
    "id": "shelter",
    "power": 1,
    "powerType": "always",
    "recipe": Object {
      "plastic": 2,
      "silicone": 2,
    },
    "shortId": "s",
    "size": Array [
      8,
      8,
    ],
    "tier": 4,
    "unlockCost": null,
  },
  "smallBattery": Object {
    "capacity": 32,
    "id": "smallBattery",
    "power": 1,
    "powerType": "powered",
    "recipe": Object {
      "zinc": 1,
    },
    "shortId": "sb",
    "size": Array [
      1,
      1,
    ],
    "tier": 1,
  },
  "smallGenerator": Object {
    "id": "smallGenerator",
    "power": 2,
    "powerType": "powered",
    "recipe": Object {
      "compound": 1,
    },
    "shortId": "sg",
    "size": Array [
      1,
      1,
    ],
    "tier": 1,
  },
  "smallSolarPanel": Object {
    "id": "smallSolarPanel",
    "power": 1,
    "powerType": "sun",
    "recipe": Object {
      "copper": 1,
    },
    "shortId": "ss",
    "size": Array [
      1,
      1,
    ],
    "tier": 1,
  },
  "smallWindTurbine": Object {
    "id": "smallWindTurbine",
    "power": 1.5,
    "powerType": "wind",
    "recipe": Object {
      "ceramic": 1,
    },
    "shortId": "swt",
    "size": Array [
      1,
      1,
    ],
    "tier": 1,
  },
  "smeltingFurnace": Object {
    "id": "smeltingFurnace",
    "power": -5,
    "recipe": Object {
      "compound": 1,
      "resin": 2,
    },
    "shortId": "sf",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 250,
  },
  "soilCentrifuge": Object {
    "id": "soilCentrifuge",
    "power": -6,
    "recipe": Object {
      "aluminum": 1,
      "compound": 2,
    },
    "shortId": "sc",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 750,
  },
  "solarArray": Object {
    "id": "solarArray",
    "power": 14,
    "recipe": Object {
      "aluminumAlloy": 1,
      "copper": 1,
      "glass": 1,
      "graphene": 1,
    },
    "shortId": "sa",
    "size": Array [
      8,
      8,
    ],
    "tier": 4,
  },
  "tradePlatform": Object {
    "id": "tradePlatform",
    "power": 0,
    "recipe": Object {
      "compound": 1,
      "iron": 1,
      "tungsten": 1,
    },
    "shortId": "tp",
    "size": Array [
      4,
      4,
    ],
    "tier": 3,
    "unlockCost": 2500,
  },
  "wreckedSolarArray": Object {
    "id": "wreckedSolarArray",
    "power": 64,
    "powerType": "sun",
    "recipe": null,
    "shortId": "wsa",
    "size": Array [
      16,
      16,
    ],
    "tier": -1,
  },
  "xlargeWindTurbine": Object {
    "id": "xlargeWindTurbine",
    "power": 17,
    "powerType": "wind",
    "recipe": Object {
      "aluminumAlloy": 1,
      "ceramic": 1,
      "graphene": 1,
      "iron": 1,
    },
    "shortId": "xlwt",
    "size": Array [
      8,
      8,
    ],
    "tier": 4,
  },
}
`;
